### PR TITLE
Issue218 show GitHub status

### LINKED ISSUE #218: Show github token status in frontend
Currently it's impossible to know whether the save button in editor tries to push or just save locally. 

~~Perhaps the `ME` query could return some boolen value whether the `context.githubToken` is set and the frontend could include an icon or some text to indicate how things are.~~

OR

Create a new query that just returns the boolean. This way the nice UserType return would be untouched for ME query and this temporary solution would be easy to remove

### PR DESCRIPTION
**Pull request by tulma95**
img

img2

Closes #218 
Backend changes: 
- New graphql query `isGithubConnected: Boolean!` which returns true or false based on if `context.user` has githubToken or not.
- Two tests for the query

Frontend changes:
- Updated @material-ui/icons
- Added `isGIthubConnected` type and query
- Add new `GHConnected` -component which is located and used in `MonacoEditor.tsx`

### CODE REVIEW

**Review by guotin — APPROVED** [review]
Nice icon and seems to indicate what is happening quite nicely. Good that the backend changes had tests written for it.

File: frontend/src/components/MonacoEditor.tsx [hunk 1]
```diff
@@ -24,9 +25,33 @@ interface DialogError {
...
+  }
+
+  return (
+    <span
+      style={{
+        marginLeft: '1rem',
+        verticalAlign: 'middle',
+      }}
+    >
+      {isGithubConnected ? githubConnected() : 'Github is not connected'}
```
Comment (jakekall) [inline frontend/src/components/MonacoEditor.tsx:44]:
```suggestion
      <Tooltip title="GitHub is connected. Saving will push to GitHub">
        <GitHub />
      </Tooltip>
    )
  }

  return (
    <span
      style={{
        marginLeft: '1rem',
        verticalAlign: 'middle',
      }}
    >
      {isGithubConnected ? githubConnected() : 'GitHub is not connected'}
```
Kinda nitpicky, but the way GitHub is written is very inconsistent. Doesn't really matter in code, but for UI it's better to keep things consistent.

Comment (tulma95) [reply frontend/src/components/MonacoEditor.tsx:44]: Nicely spotted. Which way would you prefer github to be written in?

Comment (tulma95) [reply frontend/src/components/MonacoEditor.tsx]: I noticed that you already did the changes so I could just accept it

**Review by jakekall — APPROVED** [review]

**Pull request merged**